src/main/bash/sdkman-init.sh

```diff
@@ -100,29 +107,22 @@ if [[ -z "$sdkman_curl_max_time" ]]; then sdkman_curl_max_time=10; fi
 
 # Read list of candidates and set array
 SDKMAN_CANDIDATES_CACHE="${SDKMAN_DIR}/var/candidates"
-SDKMAN_CANDIDATES_CSV=$(cat "$SDKMAN_CANDIDATES_CACHE")
+SDKMAN_CANDIDATES_CSV=$(<"$SDKMAN_CANDIDATES_CACHE")
 __sdkman_echo_debug "Setting candidates csv: $SDKMAN_CANDIDATES_CSV"
 if [[ "$zsh_shell" == 'true' ]]; then
-    SDKMAN_CANDIDATES=( ${(s:,:)SDKMAN_CANDIDATES_CSV} )
+    SDKMAN_CANDIDATES=(${(s:,:)SDKMAN_CANDIDATES_CSV})
 else
-    OLD_IFS="$IFS"
+    old_IFS="$IFS"
     IFS=","
     SDKMAN_CANDIDATES=(${SDKMAN_CANDIDATES_CSV})
-    IFS="$OLD_IFS"
+    IFS="$old_IFS"
 fi
-
-# The candidates are assigned to an array for zsh compliance, a list of words is not iterable
-# Arrays are the only way, but unfortunately zsh arrays are not backward compatible with bash
-# In bash arrays are zero index based, in zsh they are 1 based(!)
-for (( i=0; i <= ${#SDKMAN_CANDIDATES[*]}; i++ )); do
-	# Eliminate empty entries due to incompatibility
-	CANDIDATE_NAME="${SDKMAN_CANDIDATES[${i}]}"
-	CANDIDATE_DIR="${SDKMAN_CANDIDATES_DIR}/${CANDIDATE_NAME}/current"
-	if [[ -n "$CANDIDATE_NAME" && ( -h "$CANDIDATE_DIR" || -d "${CANDIDATE_DIR}" ) ]]; then
-		__sdkman_export_candidate_home "$CANDIDATE_NAME" "$CANDIDATE_DIR"
-		__sdkman_prepend_candidate_to_path "$CANDIDATE_DIR"
-	fi
-	unset CANDIDATE_NAME CANDIDATE_DIR
+for candidate_name in "${SDKMAN_CANDIDATES[@]}"; do
+    candidate_dir="${SDKMAN_CANDIDATES_DIR}/${candidate_name}/current"
+    if [[ -h "$candidate_dir" || -d "${candidate_dir}" ]]; then
```

Comment:
It's great to see that we can now iterate over an array of strings in ZSH! This wasn't possible a few years back when I wrote the original code. Your approach looks way cleaner.